Add ./examples README & basic build instructions (#236)
paritytech/ink: examples/README.md

# ink! code examples

This folder contains a set of example contracts for ink!.

Have a look at the different examples to better understand how to use ink! to build your own Substrate smart contracts.

## Preparation

For building the example smart contracts found in this folder you will need to have `cargo-contract` installed.

```

> cargo install --git https://github.com/paritytech/ink cargo-contract

Reviewer: ```suggestion
cargo install --force --git https://github.com/paritytech/ink cargo-contract
```

```

> Use the `--force` option to update to the most recent `cargo-contract` version.

Reviewer: ```suggestion
We use the `--force` to update to the most recent `cargo-contract` version.
```

Author: Is that some kind of Star Wars reference?

# ink! code examples

This folder contains a set of example contracts for ink!.

Have a look at the different examples to better understand how to use ink! to build your own Substrate smart contracts.

> Besides teaching you how to write smart contracts, this folder is also meant for testing purposes. If you want to contribute to Parity ink!, please run the `check-examples.sh` script in the `./scripts` folder before submitting your PR.

Reviewer 2: ```suggestion
Besides teaching you how to write smart contracts, this folder is also meant for testing purposes. If you want to contribute to Parity ink!, please run the `check-workspace.sh` and `check-examples.sh` script in the `./scripts` folder before submitting your PR.
```

Author: `check-workspace.sh` doesn't have anything to do with this ./examples folder, that's why I didn't mention it here. `check-examples.sh` on the other hand is using it.

It is already mentioned in the general README , that both files in the ./scripts should be run before committing a PR.

Reviewer 2: That's true but I was wondering on the whole section as a whole because we shouldn't regard it as standard practice that people start hacking on our examples and file PRs but this paragraph gave me the idea that this might be intended dev behaviour. So we might even want to consider removing this paragraph. I do not mean that devs should not file PRs for obvious improvements to our examples though.

Author: Removed.

## Preparation

For building the example smart contracts found in this folder you will need to have `cargo-contract` installed.

```
cargo install --git https://github.com/paritytech/ink cargo-contract
```

> Add `--force` option to update to the most recent `cargo-contract` version.

Reviewer 2: ```suggestion
Use the `--force` to update to the most recent `cargo-contract` version.
```

## Build example contract and generate the contracts ABI

> To build a single example and generate the contracts wasm file, navigate to the root of the smart contract and run the following command:

Reviewer 2: ```suggestion
To build a single example and generate the contracts Wasm file, navigate to the root of the smart contract and run the following command:
```

`cargo contract build`

> To generate the Application Binary Interface (ABI) of the contract, run the following command:

Reviewer 2: ```suggestion
To generate the Application Binary Interface (ABI), or metadata, of the contract, run the following command:
```

Author: I changed it to the wording suggested by @shawntabrizi for now.

`cargo contract generate-abi`

> You should now have a optimized `<contract-name>.wasm` file and a `abi.json` file in the `./target` folder of your contract.

Reviewer 2: ```suggestion
You should now have an optimized `<contract-name>.wasm` file and an `abi.json` file in the `target` folder of your contract.
```
